pennylane/io/qualtran_io.py

```diff
@@ -418,3 +583,397 @@ def compute_matrix(*params, **hyperparams):  # pylint: disable=no-self-argument
+            return f"PL{self.op.name}"
+        return "PLQfunc"
+
+
+def to_bloq(circuit, map_ops: bool = True, custom_mapping: dict = None, **kwargs):
+    """
+    Converts a PennyLane circuit or :class:`~.Operation` to the representing `Qualtran Bloq <https://qualtran.readthedocs.io/en/latest/bloqs/index.html#bloqs-library>`__.
+
+    .. note::
+        This class requires the latest version of Qualtran. We recommend installing the latest
+        release via ``pip``:
+
+        .. code-block:: console
+
+            pip install qualtran
+
+    Args:
+        circuit (QNode | Operation): A :class:`~pennylane.QNode` or an initialized PennyLane operator to be converted to a Qualtran Bloq.
+        map_ops (bool): Whether or not if the operations are mapped to a Qualtran Bloq or wrapped
+            as a ``ToBloq``. Default is ``True``.
+        custom_mapping (dict): Dictionary to specify a mapping between a PennyLane operator and a
+            Qualtran Bloq. A default mapping is used if not defined.
+
+    Returns:
+        Bloq: The Qualtran Bloq that corresponds to the given circuit or :class:`~.Operation` and
+            options.
+
+    .. seealso:: :class:`~.ToBloq`
+
+    **Example**
+
+    This example shows how to use ``qml.to_bloq``:
+
+    >>> from qualtran.drawing import show_call_graph
+    >>> from qualtran.resource_counting.generalizers import generalize_rotation_angle
+    >>> op = qml.QuantumPhaseEstimation(
+    ...     qml.RX(0.2, wires=[0]), estimation_wires=[1, 2]
+    ... )
+    >>> op_as_bloq = qml.to_bloq(op)
+    >>> graph, sigma = op_as_bloq.call_graph(generalize_rotation_angle)
+    >>> sigma
+    {Allocate(dtype=QFxp(bitsize=2, num_frac=2, signed=False), dirty=False): 1,
+    Hadamard(): 4,
+    Controlled(subbloq=Rx(angle=0.2, eps=1e-11), ctrl_spec=CtrlSpec(qdtypes=(QBit(),), cvs=(array(1),))): 3,
+    And(cv1=1, cv2=1, uncompute=True): 1,
+    And(cv1=1, cv2=1, uncompute=False): 1,
+    ZPowGate(exponent=\\phi, eps=1e-10): 1,
+    TwoBitSwap(): 1}
+
+    .. details::
+        :title: Usage Details
+
+        Some PennyLane operators don't have a direct equivalent in Qualtran. For example, in Qualtran, there
+        exists many varieties of Quantum Phase Estimation. When ``qml.to_bloq`` is called on
+        Quantum Phase Estimation, a smart default is chosen.
+
+        >>> qml.to_bloq(qml.QuantumPhaseEstimation(
+        ...     unitary=qml.RX(0.1, wires=0), estimation_wires=range(1, 5)
+        ... ))
+        TextbookQPE(unitary=Rx(angle=0.1, eps=1e-11), ctrl_state_prep=RectangularWindowState(bitsize=4), qft_inv=Adjoint(subbloq=QFTTextBook(bitsize=4, with_reverse=True)))
+
+        Note that the chosen Qualtran Bloq may not be an exact equivalent. If an exact
+        equivalent is needed, we recommend setting ``map_ops`` to False.
+        This will wrap the input PennyLane operator as a Qualtran Bloq, allowing to use Qualtran functions
+        such as ``decompose_bloq`` or ``call_graph``.
```

Comment:
Something like this?
```suggestion
        This will wrap the input PennyLane operator as a Qualtran Bloq, enabling Qualtran functions
        such as ``decompose_bloq`` or ``call_graph``, but maintaining the PennyLane decomposition definition of the operator.
```